app/controllers/welcome_controller.rb

```diff
@@ -1,5 +1,18 @@
 class WelcomeController < ApplicationController
+
   def index
+    if current_user
+      route = determine_route(current_user)
+
+      redirect_to route
+    end
+  end
 
+  def determine_route(user)
+    {
+      "admin" => "/admin",
+      "merchant" => "/merchant",
+      "default" => "/profile"
+    }[user.role]
   end
```

Comment:
If i knew how to properly comment a gif, I'd be giving you a sweet gif right here. Good work

Reply:
![](https://media2.giphy.com/media/3ohzdRyY5tYJVb8Tzq/giphy.gif?cid=5a38a5a2246c81f31c0f9f79d9093cd49918e983c88ba838&rid=giphy.gif)

Reply:
Thank you Stella!!!